Index: cups-1.2.7/scheduler/job.c
===================================================================
--- cups-1.2.7.orig/scheduler/job.c
+++ cups-1.2.7/scheduler/job.c
@@ -73,7 +73,12 @@
  */
 
 #include "cupsd.h"
+#include <sys/stat.h>
 #include <grp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
 #include <cups/backend.h>
 #include <cups/dir.h>
 
@@ -2436,6 +2441,87 @@ set_hold_until(cupsd_job_t *job, 	/* I -
   cupsdSaveJob(job);
 }
 
+static const mime_filter_t vxlpacct_filter_data = {
+    .src    = NULL,
+    .dst    = NULL,
+    .cost   = 0,
+    .filter = "lpacct_filter",
+};
+
+static int vxlpacct_enabled(void)
+{
+    struct stat sb;
+    char buf[128];
+    snprintf(buf, sizeof(buf), "%s/filter/lpacct_filter", ServerBin);
+    return stat(buf, &sb) == 0 && access(buf, X_OK) == 0;
+}
+
+static cups_array_t *vxlpacct_create_chain1(mime_t *mimedb, mime_type_t *src,
+  mime_type_t *dst, int *cost)
+{
+    cups_array_t *ret;
+    if((ret = mimeFilter(mimedb, src, dst, cost)) == NULL)
+        return NULL;
+    if(!cupsArrayInsert(ret, (void *)&vxlpacct_filter_data)) {
+        cupsdLogMessage(CUPSD_LOG_ERROR, "Unable to add vxlpacct filter - %s",
+                        strerror(errno));
+        cupsArrayDelete(ret);
+    }
+    return ret;
+}
+
+static cups_array_t *vxlpacct_create_chain2(mime_t *mimedb, mime_type_t *src,
+  mime_type_t *dst, int *cost)
+{
+    cups_array_t *chain_1, *chain_2;
+    mime_filter_t *filter;
+    int cost_1, cost_2;
+
+    chain_1 = mimeFilter(mimedb, src, mimeType(MimeDatabase, "application",
+              "vnd.cups-postscript"), &cost_1);
+    chain_2 = mimeFilter(mimedb, mimeType(MimeDatabase, "application",
+              "vnd.cups-postscript"), dst, &cost_2);
+
+    if(chain_1 == NULL || chain_2 == NULL)
+        goto err;
+    if(!cupsArrayAdd(chain_1, (void *)&vxlpacct_filter_data))
+        goto err;
+
+    for(filter = cupsArrayFirst(chain_2); filter != NULL;
+        filter = cupsArrayNext(chain_2))
+    {
+        if(!cupsArrayAdd(chain_1, filter))
+            goto err;
+    }
+
+    cupsArrayDelete(chain_2);
+    *cost = cost_1 + cost_2;
+    return chain_1;
+
+ err:
+    cupsdLogMessage(CUPSD_LOG_ERROR, "Unable to add vxlpacct filter - %s",
+                    strerror(errno));
+    cupsArrayDelete(chain_1);
+    cupsArrayDelete(chain_2);
+    return NULL;
+}
+
+static cups_array_t *vxlpacct_mimefilter(mime_t *mimedb, mime_type_t *src,
+  mime_type_t *dst, int *cost)
+{
+    if(!vxlpacct_enabled())
+        return mimeFilter(mimedb, src, dst, cost);
+
+    if(strcmp(src->super, "application") == 0 &&
+     (strcmp(src->type, "postscript") == 0 ||
+     strcmp(src->type, "vnd.cups-postscript") == 0))
+        // Input is already postscript
+        return vxlpacct_create_chain1(mimedb, src, dst, cost);
+    else
+        // Input is not postscript
+        return vxlpacct_create_chain2(mimedb, src, dst, cost);
+}
+
 
 /*
  * 'start_job()' - Start a print job.
@@ -2542,7 +2628,7 @@ start_job(cupsd_job_t     *job,		/* I - 
     * Local jobs get filtered...
     */
 
-    filters = mimeFilter(MimeDatabase, job->filetypes[job->current_file],
+    filters = vxlpacct_mimefilter(MimeDatabase, job->filetypes[job->current_file],
                          printer->filetype, &(job->cost));
 
     if (!filters)
